tests/bgp/test_bgp_speaker.py

```diff
@@ -199,3 +224,68 @@ def test_bgp_speaker_announce_routes(common_setup_teardown, testbed, duthost, pt
                         "testbed_mtu": mtu },
                 log_file="/tmp/bgp_speaker_test.FibTest.log",
                 socket_recv_size=16384)
+
+
+@pytest.mark.parametrize("ipv4, ipv6, mtu", [pytest.param(False, True, 1514)])
+def test_bgp_speaker_announce_routes_v6(common_setup_teardown, testbed, duthost, ptfhost, ipv4, ipv6, mtu, collect_techsupport):
```

Comment:
i can still see duplicated code between v6 and v4 annouce routes. I think you can have a common function that abstract the check logic and let both functions call the common function with the different parameter.

Reply:
Done